gp-includes/things/translation.php

```diff
@@ -422,19 +422,51 @@ public function for_translation( $project, $translation_set, $page, $filters = a
 
 		$join_where = implode( ' AND ', $join_where );
 		if ( $join_where ) {
-			$join_where = 'AND '.$join_where;
+			$join_where = 'AND ' . $join_where;
 		}
 
-		$sql_sort = sprintf( $sort_by, $sort_how );
-
+		$fields = 't.*, o.*, t.id as id, o.id as original_id, t.status as translation_status, o.status as original_status, t.date_added as translation_added, o.date_added as original_added';
+		$join = "$join_type JOIN {$wpdb->gp_translations} AS t ON o.id = t.original_id AND t.translation_set_id = " . (int) $translation_set->id;
+		$orderby = sprintf( $sort_by, $sort_how );
 		$limit = $this->sql_limit_for_paging( $page, $this->per_page );
 
+		/**
+		 * Filters the 'for_translation' query SQL clauses.
+		 *
+		 * @since 2.3
+		 *
+		 * @param array              $pieces          Terms query SQL clauses.
+		 * @param GP_Translation_Set $translation_set The translation set object being queried.
+		 * @param array              $filters         An array of search filters.
+		 * @param array              $sort            An array of sort settings.
+		 */
+		$clauses = apply_filters( 'gp_for_translation_clauses', compact( 'fields', 'join', 'join_where', 'where', 'orderby', 'limit' ), $translation_set, $filters, $sort );
+
+		$fields = isset( $clauses['fields'] ) ? $clauses['fields'] : '';
```

Comment:
It's unlikely imo that a plugin would return an incomplete clauses object intentionally, but I agree that the current default value doesn't make sense if the intention is to prevent a problem. Fixing.